### PR TITLE
fix: proper repo location on autoupdate

### LINKED ISSUE #93: Persistent SSL certificate issues
Follow up from #83

Hi @RobPasMue 
Thanks for quick fix.
1. This works for downloading now, I cannot install Python because of `Admin Rights`. That is fine for now.
2. I have also encountered  SSL problem with updater 

grafik

I am not really sure where these SSL problems are arising from exactly, because I can access everything over the browser without any problems.

_Originally posted by @kmahajan-cadfem in https://github.com/pyansys/python-installer-qt-gui/issues/83#issuecomment-1527736318_

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says. I will also try to fix #93 

* Refer to ``ansys/python-installer-qt-gui`` instead of ``pyansys/python-installer-qt-gui``
* Wrap pygithub query with ``try...except``. Closes #93

### CODE REVIEW

**Comment by RobPasMue** [comment]
New message box in case check for update fails:

image

File: src/ansys/tools/installer/main.py [hunk 1]
```diff
@@ -247,13 +248,34 @@ def _close(self):
...
 
         LOG.debug(f"Currently installed version: {cur_ver}")
         LOG.debug(f"Latest version: {ver}")
 
-        if ver > cur_ver:
+        if not ver:
+            # Error occurred while requesting version... update check
+            # cannot be automated. Referring to source.
+            LOG.debug(
+                "Update check cannot be automatically performed. Showing info message"
```
Comment (RobPasMue) [inline src/ansys/tools/installer/main.py:266]:
```suggestion
                "Update check cannot be automatically performed. Showing info message."
```

File: src/ansys/tools/installer/main.py [hunk 2]
```diff
@@ -247,13 +248,34 @@ def _close(self):
     def check_for_updates(self):
         """Check for Ansys Python Manager application updates."""
         LOG.debug("Checking for updates")
-        (ver, url) = query_gh_latest_release()
+        try:
+            (ver, url) = query_gh_latest_release()
+        except:
+            LOG.info(f"Problem requesting version... ")
```
Comment (RobPasMue) [inline src/ansys/tools/installer/main.py:254]:
```suggestion
            LOG.info("Problem requesting version... ")
```

**Comment by kmahajan-cadfem** [comment]
I am not able to run the UI from source code because again SSL errors from [pypi api](api.pypi.org).
I do understand that this is necessary, but would it be possible to maintain a list of versions(of all the possible versions from) in the package itself, then at least the UI would open, in my case the UI just does not open and user does not know whats wrong.

`--console` flag does not work with the exe, I had to run the __main__.py file

```shell
2023-08-28 11:32:26,874 - DEBUG - Populating the table with virtual environments.
2023-08-28 11:32:26,881 - DEBUG - Populating the table with python and conda forge versions.
2023-08-28 11:32:26,881 - DEBUG - Found the following unified Ansys installation versions: {232: 'C:\\Program Files\\ANSYS Inc\\v232', 231: 'C:\\Program Files\\ANSYS Inc\\v231', 221: 'C:\\Program Files\\ANSYS Inc\\v221'}
2023-08-28 11:32:27,255 - DEBUG - Starting new HTTPS connection (1): pypi.python.org:443
Traceback (most recent call last):
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connectionpool.py", line 467, in _make_request
    self._validate_conn(conn)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connectionpool.py", line 1092, in _validate_conn    conn.connect()
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connection.py", line 642, in connect
    sock_and_verified = _ssl_wrap_socket_and_match_hostname(
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connection.py", line 783, in _ssl_wrap_socket_and_match_hostname
    ssl_sock = ssl_wrap_socket(
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\util\ssl_.py", line 469, in ssl_wrap_socket     
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\util\ssl_.py", line 513, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "C:\Program Files\ANSYS Inc\v232\commonfiles\CPython\3_10\winx64\Release\python\lib\ssl.py", line 513, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Program Files\ANSYS Inc\v232\commonfiles\CPython\3_10\winx64\Release\python\lib\ssl.py", line 1071, in _create
    self.do_handshake()
  File "C:\Program Files\ANSYS Inc\v232\commonfiles\CPython\3_10\winx64\Release\python\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)   

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connectionpool.py", line 790, in urlopen        
    response = self._make_request(
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connectionpool.py", line 491, in _make_request  
    raise new_e
urllib3.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)    

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\connectionpool.py", line 844, in urlopen        
    retries = retries.increment(
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\urllib3\util\retry.py", line 515, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='pypi.python.org', port=443): Max retries exceeded with url: /pypi/pyansys/json (Caused by 
SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\src\ansys\tools\installer\__main__.py", line 6, in <module>
    open_gui()
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\ansys\tools\installer\main.py", line 680, in open_gui   
    window = AnsysPythonInstaller()
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\ansys\tools\installer\main.py", line 123, in __init__   
    self.installed_table_tab = InstalledTab(self)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\ansys\tools\installer\installed_table.py", line 281, in 
__init__
    self.update_package_combo(0)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\ansys\tools\installer\installed_table.py", line 362, in 
update_package_combo
    self._cached_versions[package_name] = get_pkg_versions(package_name)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\ansys\tools\installer\common.py", line 88, in get_pkg_versions
    releases = json.loads(requests.get(url).content)["releases"]
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "D:\users\mkanchan\Work\pyansys\source\python-installer-qt-gui\.venv\lib\site-packages\requests\adapters.py", line 517, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='pypi.python.org', port=443): Max retries exceeded with url: /pypi/pyansys/json (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)')))
```

**Comment by RobPasMue** [comment]
@kmahajan-cadfem - yep got it...

Keeping track of a hardcoded list of versions might be a high-burden to pay for... What I can do, though, is to avoid the failure and just show the list of packages. That way, if you then request the installation of a certain package, you would just get the latest version of it (i.e. it's like instead of doing ``pip install pyansys==XX.YY.ZZ`` you just ran ``pip install pyansys``.

Would that work?

**Comment by kmahajan-cadfem** [comment]
> @kmahajan-cadfem - yep got it...
> 
> Keeping track of a hardcoded list of versions might be a high-burden to pay for... What I can do, though, is to avoid the failure and just show the list of packages. That way, if you then request the installation of a certain package, you would just get the latest version of it (i.e. it's like instead of doing `pip install pyansys==XX.YY.ZZ` you just ran `pip install pyansys`).
> 
> Would that work?


I agree, maintaining list is a manual work and would not be efficient.

but  just list without versions(i think this is already there) would be a great advantage and it is better to tell user what is failing or it should not fail to open.

once the user at least reaches to the UI , `venvs` creation should not be a problem, the last remaining bit of installation of `pip` packages could be done at later point or through console.

**Comment by RobPasMue** [comment]
Just did some implementation "simulating" your case (no connection to the PyPI API). Let me know what you think. I'll upload the changes

[pypi-api.webm](https://github.com/ansys/python-installer-qt-gui/assets/37798125/c05b3cc4-aed2-492b-84c0-219fec4eb873)

**Comment by kmahajan-cadfem** [comment]
This looks great 👍

**Comment by RobPasMue** [comment]
Awesome, thanks @kmahajan-cadfem! Please review and approve whenever you have the chance so that we can merge it in that case. Glad it solved your issues!

**Review by kmahajan-cadfem — APPROVED** [review]

**Pull request merged**